src/en/administration/from_scratch.md

```diff
@@ -0,0 +1,886 @@
+cp target/release/pict-rs .
+```
+
+restart pictrs
+
+---
+
+## Index: scripts
+
+### config script
+
+config/config.hjson example
+
+```json
+{
+  database: {
+    user: "lemmy"
+    password: "whatever"
+    host: "localhost"
+    port: 5432
+    database: "lemmy"
+    pool_size: 5
+  }
+  hostname: "lemmy.ca"
+  bind: "127.0.0.1"
+  port: 8536
+  docs_dir: "/home/lemmy/lemmy/docs/book"
+  pictrs_url: "http://localhost:9000"
+  iframely_url: "http://localhost:8061"
+  federation: {
+    enabled: true
+    allowed_instances: ""
+    blocked_instances: ""
+  }
+  email: {
+    smtp_server: "localhost:25"
+    smtp_from_address: "lemmy@lemmy.ca"
+    use_tls: false
+  }
+}
+```
+
+### init scripts
+
+##### init/iframely
+
+```bash
+#!/sbin/openrc-run
+
+name="Iframely Daemon"
+
+depend() {
+        need localmount
+        need net
+}
+
+start() {
+        ebegin "Starting Iframely"
+        start-stop-daemon --start --background --make-pidfile --user iframely --group iframely --pidfile /home/iframely/iframely.pid --chdir /home/iframely/iframely -3 /usr/bin/logger -4 /usr/bin/logger --exec node -- server
+        eend $?
+}
+
+stop() {
+        start-stop-daemon --stop --signal TERM --pidfile /home/iframely/iframely.pid
+        eend $?
+}
+```
+
+##### init/lemmy
+
+```bash
+#!/sbin/openrc-run
+
+name="Lemmy Backend"
+
+depend() {
+        need localmount
+        need net
+}
+
+start() {
+        ebegin "Starting Lemmy"
+        start-stop-daemon --start --background --make-pidfile --user lemmy --group lemmy --pidfile /home/lemmy/lemmy.pid --chdir /home/lemmy/lemmy -3 /usr/bin/logger -4 /usr/bin/logger --exec ~lemmy/lemmy/target/release/lemmy_server
+        eend $?
+}
+
+stop() {
+        start-stop-daemon --stop --signal TERM --pidfile /home/lemmy/lemmy.pid
+        eend $?
+}
+```
+
+##### init/lemmy-ui
+
+```bash
+#!/sbin/openrc-run
+
+name="Lemmy UI"
+
+depend() {
+        need localmount
+        need net
+}
+
+start() {
+        ebegin "Starting Lemmy UI"
+        start-stop-daemon --start --background --make-pidfile --user lemmy --group lemmy --pidfile /home/lemmy/lemmy-ui.pid --chdir /home/lemmy/lemmy-ui -3 /usr/bin/logger -4 /usr/bin/logger --exec node dist/js/server.js --env LEMMY_INTERNAL_HOST=127.0.0.1:8536 --env LEMMY_EXTERNAL_HOST=lemmy.ca --env LEMMY_HTTPS=true
+        eend $?
+}
+```
+
+##### init/pict-rs
+
+```bash
+#!/sbin/openrc-run
+
+name="pict-rs Daemon"
+
+depend() {
+        need localmount
+        need net
+}
+
+start() {
+        ebegin "Starting pictrs"
+        start-stop-daemon --start --background --make-pidfile --user pictrs --group pictrs --pidfile /home/pictrs/pictrs.pid --chdir /home/pictrs/pict-rs -3 /usr/bin/logger -4 /usr/bin/logger --exec /home/pictrs/pict-rs/pict-rs -- -a 127.0.0.1:9000 -p ~pictrs/pictrs-data
+        eend $?
+}
+
+stop() {
+        start-stop-daemon --stop --signal TERM --pidfile /home/pictrs/pictrs.pid
+        eend $?
+}
+```
+
+### iframely script
+
+iframely.local.js example
```

Comment:
We are simply using the default config for iframely, no need to include that all here.